automation/roles/postgresql-privs/tasks/main.yml

```
@@ -0,0 +1,22 @@
+---
+
+- name: Grant/revoke privileges on objects
+  community.postgresql.postgresql_privs:
+    roles: "{{ item.role }}"
+    privs: "{{ item.privs }}"
+    type: "{{ item.type }}"
+    objs: "{{ item.objs }}"
+    schema: "{{ item.schema | default(omit) }}"
+    db: "{{ item.db }}"
+    state: "{{ item.state | default('present') }}"
+    login_host: "127.0.0.1"
+    login_port: "{{ postgresql_port }}"
+    login_user: "{{ patroni_superuser_username }}"
+    login_password: "{{ patroni_superuser_password }}"
+  ignore_errors: true  # noqa ignore-errors
+  loop: "{{ postgresql_privs | flatten(1) }}"
+  when:
+    - postgresql_privs | default('') | length > 0
+    - item.role | default('') | length > 0
```

```suggestion
    - item.role | default('') | length > 0
    - item.db | default('') | length > 0
```

or so, in this case, the task will not be completed at all if the necessary parameters are not specified, and not errors. Otherwise, we will need to implement the checks in the pre-checks. 

I think it is possible to describe in the comment that "db" and "role" are required.

---

Done.